components/style/stylist.rs

```diff
@@ -389,6 +389,7 @@ impl Stylist {
             false
         }
         self.is_device_dirty |= stylesheets.iter().any(|stylesheet| {
+            stylesheet.media.read().evaluate(&self.device) ||
```

Comment:
This check is wrong, we need to check whether the media query _changed_, not only if it evaluates to true. Otherwise we might mark the device dirty when we shouldn't.